x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/processors_header.tsx

```diff
@@ -4,68 +4,146 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { FunctionComponent } from 'react';
+import React, { FunctionComponent, useState } from 'react';
 import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiText, EuiTitle } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n/react';
 
 import { useKibana } from '../../../shared_imports';
 
+import {
+  usePipelineProcessorsContext,
+  useTestPipelineContext,
+} from '../pipeline_processors_editor/context';
+
 import {
   LoadFromJsonButton,
   OnDoneLoadJsonHandler,
-  TestPipelineButton,
+  TestOutputButton,
+  DocumentsDropdown,
+  AddDocumentsButton,
+  TestPipelineFlyout,
+  TestPipelineFlyoutTab,
 } from '../pipeline_processors_editor';
 
 export interface Props {
   onLoadJson: OnDoneLoadJsonHandler;
 }
 
 export const ProcessorsHeader: FunctionComponent<Props> = ({ onLoadJson }) => {
+  const {
+    state: { processors },
+  } = usePipelineProcessorsContext();
+  const { testPipelineData, setCurrentTestPipelineData } = useTestPipelineContext();
```

Comment:
Great point. I agree; I think this would be cleaner. I think I may have started in this direction, but can't remember now why I pivoted 🤔 . I'll look into refactoring this.

Reply:
I ended up creating a component called `TestPipelineActions` that renders the add documents button/documents dropdown and the output button. The pipeline processor editor exports this as well as the `DocumentsDropdown`, since the dropdown is also reused in the output tab of each processor. Let me know what you think!